### ISSUE TITLE
hdevtools: <socket: 4>: hGetLine: end of file

### ISSUE THREAD

**Issue by bonds** [hdevtools/hdevtools#50]
When checking a file with the Turtle library imported, I get an error. Here's a minimal example:

````
~  [1] ❯❯❯ cat blah.hs                                                      Sun 2017-04-30 16:59:18 -0700
import qualified Turtle as TU
main = putStrLn "hello, world"
~ ❯❯❯ hdevtools check -d blah.hs                                            Sun 2017-04-30 16:59:25 -0700
Cabal file: Nothing
Run from outside a project, using implicit global project config
Stack file: Just "/home/scott/.stack/global-project/stack.yaml"
Socket file: ".hdevtools.sock"
hdevtools: <socket: 4>: hGetLine: end of file
````

If I remove the Turtle import, I can check the file just fine. If I add other, non-Turtle imports, it also checks fine.

The Turtle library seems to be working ok:

````
~  [1] ❯❯❯ ghci                                                             Sun 2017-04-30 16:59:31 -0700
GHCi, version 7.10.3: http://www.haskell.org/ghc/  :? for help
Ok, modules loaded: none.
ghci> import Turtle
ghci> echo "hello there"
hello there
ghci>
Leaving GHCi.
````

I'm using:

````
~ ❯❯❯ hdevtools --version                                                   Sun 2017-04-30 17:01:42 -0700
hdevtools: version 0.1.5.0 (ghc-7.10.3-x86_64-openbsd, cabal-1.22.8.0)
~ ❯❯❯ stack --version                                                       Sun 2017-04-30 17:02:19 -0700
Version 1.1.2 x86_64 hpack-0.14.1
~ ❯❯❯ ghc --version                                                         Sun 2017-04-30 17:02:23 -0700
The Glorious Glasgow Haskell Compilation System, version 7.10.3
~ ❯❯❯                               
````

**Comment by bonds** [comment]
cc @Gabriel439

**Comment by bonds** [comment]
````stack exec -- hdevtools check -d blah.hs```` works fine, I gather it couldn't find the Turtle lib without that. Not a bug? I should just use stack exec?

**Comment by Gabriel439** [comment]
Check to see if `turtle` is in the installed `package` list by running `ghc-pkg list` (with and without `stack`)

**Comment by ch1bo** [comment]
Hdevtools selects the ghc and package databases depending on found cabal and stack files. Running with -d enables debug output, shows that it couldn't find any from your test path and probably uses the system ghc with packages installed in just the system package database. Wrapping everything with stack exec provides hdevtools with a stack installed ghc and package database.

If you expected hdevtools to use the ghc/packages from your global stack.yaml, we could identify that as an improvement here.

Can you show me your `~/.stack/global/stack.yaml` and the output of `stack exec -- which ghc`?

⁣Sent from TypeApp ​

**Comment by bonds** [comment]
````
~ ❯❯❯ ghc-pkg list | grep turtle                                            Mon 2017-05-01 05:35:21 -0700
~  [1] ❯❯❯ stack exec -- ghc-pkg list | grep turtle                         Mon 2017-05-01 05:35:28 -0700
Run from outside a project, using implicit global project config
    turtle-1.2.8
~ ❯❯❯ cat ~/.stack/global-project/stack.yaml                                Mon 2017-05-01 05:35:40 -0700
# This is the implicit global project's config file, which is only used when
# 'stack' is run outside of a real project.  Settings here do _not_ act as
# defaults for all projects.  To change stack's default settings, edit
# '/home/scott/.stack/config.yaml' instead.
#
# For more information about stack's configuration, see
# http://docs.haskellstack.org/en/stable/yaml_configuration/
#
flags: {}
extra-package-dbs: []
packages: []
extra-deps:
- X11-xft-0.3.1 # for monad-contrib
- cond-0.4.1.1 # for hbro
- gogol-core-0.1.0 # for gogol-youtube
- cereal-text-0.1.0.2 # for chatter
- fullstop-0.1.4 # for chatter
- tokenize-0.3.0 # for chatter
resolver: lts-6.31
~ ❯❯❯ stack exec -- which ghc                                               Mon 2017-05-01 05:36:23 -0700
Run from outside a project, using implicit global project config
Using resolver: lts-6.31 from implicit global project's config file: /home/scott/.stack/global-project/stack.yaml
/usr/local/bin/ghc
~ ❯❯❯ ghc-pkg list | grep -E '^/'                                      Mon 2017-05-01 05:43:46 -0700
/usr/local/lib/ghc/package.conf.d:
/home/scott/.ghc/x86_64-openbsd-7.10.3/package.conf.d:
~ ❯❯❯ stack exec -- ghc-pkg list | grep -E '^/'                             Mon 2017-05-01 05:43:53 -0700
Run from outside a project, using implicit global project config
/usr/local/lib/ghc/package.conf.d:
/home/scott/.stack/snapshots/x86_64-openbsd/lts-6.31/7.10.3/pkgdb:
/home/scott/.stack/global-project/.stack-work/install/x86_64-openbsd/lts-6.31/7.10.3/pkgdb:

````

Sorry to drag you in @Gabriel439, doesn't seem to be Turtle related, feel free to unsubscribe! 

Thanks @ch1bo. Perhaps it would be worth improving the error message, if possible? "hdevtools: <socket: 4>: hGetLine: end of file" leaves me perplexed, but something like "hdevtools tried to find the 'Turtle' library but could not...perhaps you need to run stack exec -- hdevtools check instead" would have pointed me in the right direction. 

BTW, on my machine, I only have the system installed ghc, I don't have any project based ghc bins.

**Comment by ch1bo** [comment]
Yeah, what happens is that the server part of hdevtools errors and the CLI can't communicate with it anymore. If you run the server in non-daemon: `hdevtools admin start-server -n` (or so) you can see the actual error.

⁣Sent from TypeApp ​

**Comment by bonds** [comment]
Ah. So, the server would need to handle the exception, send the error message up to the client, and only then rethrow...or something like that.

**Comment by ch1bo** [comment]
Open issue with cryptic error message should be fixed with #61
